lib/semantics/resolve-names.cc

```diff
@@ -2086,14 +2090,17 @@ void ModuleVisitor::BeginModule(const parser::Name &name, bool isSubmodule) {
 // If an error occurs, report it and return nullptr.
 Scope *ModuleVisitor::FindModule(const parser::Name &name, Scope *ancestor) {
   ModFileReader reader{context()};
-  auto *scope{reader.Read(name.source, ancestor)};
+  Scope *scope{reader.Read(name.source, ancestor)};
   if (!scope) {
     return nullptr;
   }
   if (scope->kind() != Scope::Kind::Module) {
     Say(name, "'%s' is not a module"_err_en_US);
     return nullptr;
   }
+  if (DoesScopeContain(scope, currScope())) {
+    Say(name, "Module '%s' cannot USE itself."_err_en_US);
```

Comment:
Should return `nullptr` here too?

Reply:
I worried that doing so would elicit needless error messages in the following code.